os/drivers/lcd/lcd_framebuffer.c

```diff
@@ -234,45 +241,46 @@ static int lcdfb_update(FAR struct lcdfb_dev_s *priv,
 
 #ifdef CONFIG_LCD_DMA_SUPPORT
 	if (pinfo->putdma) {
-		//send buffer with dma , need to convert buffer
-		int y, w;
-		int bytepp = pinfo->bpp / 8;
-		int height = endy - starty + 1;
-
-		int offset = 0;
-		uint8_t *dmabuff = (uint8_t *)malloc(width * height * bytepp);
+		/* Send buffer with dma , need to convert buffer 
+		 * buffer for each line is byte aligned.
+		 */		
+		bytespl = (width * pinfo->bpp + 7) >> 3;
+		height = endy - starty + 1;
+		
+		uint8_t *dmabuff = (uint8_t *)malloc(height * bytespl);
```

Comment:
TizenRT's malloc and free have a lot of costs. Do you have any other method to remove malloc / free?

Reply:
I amended another refinement for full width update, but there isn't a suitable way to remove malloc / free here, I think a long time occupied buffer which have big size is not a good idea for the updating procedure which is called on command, although we may apply it for performance optimizing.